### PR TITLE
diopser: init at unstable-2021-5-13

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123931` at 96401204 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>diopser</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/audio/diopser/default.nix:66:3:
```
   |
66 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/audio/diopser/default.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{ lib, stdenv, fetchFromGitHub, cmake, pkg-config,
+  libjack2, alsaLib , freetype, libX11, libXrandr, libXinerama, libXext, libXcursor
+}:
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/diopser/default.nix:3]:
```suggestion
{ lib, stdenv, fetchFromGitHub, cmake, pkg-config
, libjack2, alsaLib , freetype, libX11, libXrandr, libXinerama, libXext, libXcursor
}:
```

File: pkgs/applications/audio/diopser/default.nix [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    cd "$sourceRoot"
+    cp -R --no-preserve=mode,ownership ${function2.src} function2
+    cp -R --no-preserve=mode,ownership ${juce.src} JUCE
+    sed -i 's@CPMAddPackage("gh:juce-framework/JUCE.*@add_subdirectory(JUCE)@g' CMakeLists.txt
+    sed -i 's@CPMAddPackage("gh:Naios/function2.*@add_subdirectory(function2)@g' CMakeLists.txt
+    )'';
+
+  postPatch = ''
+    patchShebangs .
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/diopser/default.nix:64]:
```suggestion
  src = fetchFromGitHub {
    owner = "robbert-vdh";
    repo = pname;
    fetchSubmodules = true;
    rev = "d5fdc92f1caf5a828e071dac99e106e58f06d84d";
    sha256 = "06y1h895yxh44gp4vxzrna59lf7nlfw7aacd3kk4l1g56jhy9pdx";
  };

  postUnpack = ''(
    cd "$sourceRoot"
    cp -R --no-preserve=mode,ownership ${function2.src} function2
    cp -R --no-preserve=mode,ownership ${juce.src} JUCE
    sed -i 's@CPMAddPackage("gh:juce-framework/JUCE.*@add_subdirectory(JUCE)@g' CMakeLists.txt
    sed -i 's@CPMAddPackage("gh:Naios/function2.*@add_subdirectory(function2)@g' CMakeLists.txt
    )'';

  postPatch = ''
    patchShebangs .
  '';

  nativeBuildInputs = [ cmake pkg-config ];

  buildInputs = [
    libjack2 alsaLib freetype libX11 libXrandr libXinerama libXext
    libXcursor
  ];

  cmakeFlags = [
    "-DCMAKE_AR=${stdenv.cc.cc}/bin/gcc-ar"
    "-DCMAKE_RANLIB=${stdenv.cc.cc}/bin/gcc-ranlib"
  ];
```

Comment (SuperSandro2000) [reply pkgs/applications/audio/diopser/default.nix:64]: Thats the common order for those phases.

File: pkgs/applications/audio/diopser/default.nix [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    src = fetchFromGitHub {
+      owner = "Naios";
+      repo = "function2";
+      rev = version;
+      hash = "sha256-JceZU8ZvtYhFheh8BjMvjjZty4hcYxHEK+IIo5X4eSk=";
+    };
+  };
+
+  juce = rec {
+    version = "1a5fb5992a1a4e28e998708ed8dce2cc864a30d7";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/diopser/default.nix:19]:
```suggestion
  juce = {
    version = "unstable-XXXX-XX-XX";
```

replace XXXX-XX-XX with the date of the commit.

File: pkgs/applications/audio/diopser/default.nix [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  };
+
+  postUnpack = ''(
+    cd "$sourceRoot"
+    cp -R --no-preserve=mode,ownership ${function2.src} function2
+    cp -R --no-preserve=mode,ownership ${juce.src} JUCE
+    sed -i 's@CPMAddPackage("gh:juce-framework/JUCE.*@add_subdirectory(JUCE)@g' CMakeLists.txt
+    sed -i 's@CPMAddPackage("gh:Naios/function2.*@add_subdirectory(function2)@g' CMakeLists.txt
+    patchShebangs .
+    )'';
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/diopser/default.nix:48]:
```suggestion
  postUnpack = ''
    (
      cd "$sourceRoot"
      cp -r --no-preserve=mode,ownership ${function2.src} function2
      cp -r --no-preserve=mode,ownership ${juce.src} JUCE
      sed -i 's@CPMAddPackage("gh:juce-framework/JUCE.*@add_subdirectory(JUCE)@g' CMakeLists.txt
      sed -i 's@CPMAddPackage("gh:Naios/function2.*@add_subdirectory(function2)@g' CMakeLists.txt
      patchShebangs .
    )
  '';
```

**Pull request merged**